BannerlordTwitch/BannerlordTwitch/Api/BLTApi.cs

```diff
@@ -0,0 +1,148 @@
+        public readonly int port = 9000;
+        private static HttpListener listener;
+        public BLTApi()
+        {
+            _ = RunServerAsync();
+            bltApiUrls.Add(new BLTApiUrl("GET", "", "Show all registered api path", OnApiVisit));
+            bltApiUrls.Add(new BLTApiUrl("GET", "command", "Used to execute commands", OnCommand));
+        }
+
+        public async Task RunServerAsync()
+        {
+            Trace.WriteLine("Api server starting");
+            bool runServer = true;
+            listener = new HttpListener();
+            listener.Prefixes.Add("http://localhost:"+port+"/api/");
+            listener.Start();
+
+            while (runServer)
+            {
+                HttpListenerContext ctx = await listener.GetContextAsync();
+
+                HttpListenerRequest req = ctx.Request;
+                HttpListenerResponse resp = ctx.Response;
+
+                OnUrlRequest(req, resp);
+            }
+        }
+
+        public void OnUrlRequest(HttpListenerRequest req, HttpListenerResponse resp)
+        {
+            Trace.WriteLine("[VISITED]" + req.Url.ToString());
+            bltApiUrls.ForEach(bltApiUrl =>
+            {
+                if (
+                    "http://localhost:"+port+"/api/" + bltApiUrl.url == req.Url.ToString().Split('?')[0] || 
+                    "http://localhost:" + port + "/api/" + bltApiUrl.url + "/" == req.Url.ToString().Split('?')[0] || 
+                    "http://localhost:"+port+ "/api" + bltApiUrl.url == req.Url.ToString().Split('?')[0]
+                )
+                {
+                    Trace.WriteLine("[EXEC VISITED]" + bltApiUrl.url);
+                    try { 
+                        bltApiUrl.onVisit.Invoke(req,resp);
+                    }catch(Exception e){
+                        Log.LogFeedSystem("Issue on visit api urls from BLTApiUrl : " + bltApiUrl.url);
+                    }
+                }
+            });
+        }
+
+        public void OnApiVisit(HttpListenerRequest req, HttpListenerResponse resp)
+        {
+            dynamic json = new ExpandoObject();
+            string url = HttpUtility.UrlDecode(req.Url.ToString());
+            json.url = url;
+            SendJSON(resp, json);
+        }
+        public void OnCommand(HttpListenerRequest req, HttpListenerResponse resp)
+        {
+            dynamic json = new ExpandoObject();
+
+            string cmdName = "";
+            string pseudo = "";
+            string args = "";
+
+            try
+            {
+                var splittedUrl = req.Url.ToString().Split('?');
+                //if url have option
+                var splittedOption = splittedUrl[1].Split('&');
+                //foreach option
+                for (int i = 0; i < splittedOption.Length; i++)
+                {
+                    string[] option = splittedOption[i].Split('=');
+                    //check key & value
+                    if (option[0] != null && option[1] != null)
```

Comment:
If you split a string, none of the returned values can be null, and if there was only one part then accessing index [1] will throw an exception.
Try this instead:
```c#
var keyValuePairs = HttpUtility.UrlDecode(splittedUrl)
    .Split('&')
    .Select(kv => kv.Split('='))
    .ToDictionary(kv => kv[0], kv => kv.Count() > 1 ? kv[1] : null);
keyValuePairs.TryGetValue("cmd", out cmdName); // will be null if not found
keyValuePairs.TryGetValue("pseudo", out pseudo); // will be null if not found
keyValuePairs.TryGetValue("args", out args); // will be null if not found
```